x/gamm/keeper/swap.go

```diff
@@ -9,26 +9,32 @@ import (
 	"github.com/osmosis-labs/osmosis/v7/x/gamm/types"
 )
 
+// SwapExactAmountIn attempts to swap one asset, tokenIn, for another asset
+// denominated via tokenOutDenom through a pool denoted by poolId specifying that
+// tokenOutMinAmount must be returned in the resulting asset returning an error
+// upon failure. Upon success, the resulting tokens swapped for are returned. A
+// swap fee is applied determined by the pool's parameters.
 func (k Keeper) SwapExactAmountIn(
 	ctx sdk.Context,
 	sender sdk.AccAddress,
 	poolId uint64,
 	tokenIn sdk.Coin,
 	tokenOutDenom string,
 	tokenOutMinAmount sdk.Int,
-) (tokenOutAmount sdk.Int, err error) {
```

Comment:
Nit: Can we keep this if possible? I think this way's way less confusing

Reply:
Yeah, but here's the thing with named return variables. They're very prone to bugs if you're not careful.

E.g.: https://github.com/osmosis-labs/osmosis/pull/1174/files#diff-14148a5e967be907c5322344ffb4f10ebd6e85b59a995c9e6cc39409235f41de

Can you spot the bug? :p

Reply:
Wow I actually can't. What's the bug here?

Reply:
`shareOutAmount` was never actually set! So you'd always be returning the zero value. This is why named returns kind be bug-prone if you're not careful. I wish there was a linter for this because I do agree that named returns offer helpful additional context. I'm happy to keep the original change, but we just have to be very very careful.

I wouldn't be surprised if there's other methods that have this same bug.

Reply:
Wow. I'm seriously surprised. Yeah I think we should make an issue to go over / I can take some extra time to go over other code that might suspectedly have the same bug. 

I'm wondering how this was working before?